src/components/description_list/description_list_types.ts

```diff
@@ -43,6 +43,11 @@ export interface EuiDescriptionListProps {
    * Vertical spacing added between `EuiDescriptionList` elements
    */
   gutterSize?: EuiDescriptionListGutterSizes;
+  /**
+   * Only applies to `column`/`responsiveColumn` description lists.
+   * Allows customizing the horizontal gap between columns.
+   */
+  columnGap?: EuiDescriptionListColumnGapSizes;
```

Comment:
[VERY optional comment, just me thinking out loud]

I wonder if we should keep the naming of this prop in line with the existing `gutterSize` prop, aka

```suggestion
  columnGutterSize?: EuiDescriptionListColumnGutterSizes;
```

This feels more consistent to me with the existing naming we use for props like this, although I don't feel super strongly about this and am totally willing to be overridden on it.

Reply:
That's a good point. We should be more intentional and consistent with our naming. 

Should we update gutter props across `EUI` to `rowGutterSize` and `columnGutterSize`?

Reply:
I really like the specificity of `rowGutterSize`, but it's worth noting that renaming `gutterSize` would be a breaking change (although a fairly easy one, so I'm super down for it 🙌).

If you're good with this PR including a breaking change, I'm all for it!

Reply:
Resolved in https://github.com/elastic/eui/pull/7062/commits/06ba99a70374e92c084c3e6834f806c05b4d765e